src/store/eventing/middleware.ts

```diff
@@ -0,0 +1,30 @@
+import { Action, Dispatch, Middleware, MiddlewareAPI } from '@reduxjs/toolkit';
+import { toggleAnnotationModeAction } from '../common/actions';
+import handleModeEvents from './modes';
+import handleVisibilityEvents from './visibility';
+import { setVisibilityAction } from '../common';
+import { EventHandlerEntry } from './types';
+
+// Array of event handlers based on redux action. To add handling for new events add an entry keyed by action
+const eventHandlers: EventHandlerEntry[] = [
```

Comment:
should these be passed in as an argument to the createStore to make testing easier?

Reply:
Good idea